hudi-client/hudi-spark-client/src/main/java/org/apache/hudi/client/SparkRDDTableServiceClient.java

```diff
@@ -67,8 +68,9 @@ public class SparkRDDTableServiceClient<T> extends BaseHoodieTableServiceClient<
 
   protected SparkRDDTableServiceClient(HoodieEngineContext context,
                                        HoodieWriteConfig clientConfig,
-                                       Option<EmbeddedTimelineService> timelineService) {
-    super(context, clientConfig, timelineService);
+                                       Option<EmbeddedTimelineService> timelineService,
+                                       Option<TransactionManager> txnManager) {
```

Comment:
Not sure if it's a good idea to expose this in the constructor, as the client can escape the transaction manager and cause correctness issues in concurrency control.